src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -211,7 +212,20 @@ CreateSingleTaskRouterPlan(Query *originalQuery, Query *query,
 		modifyTask = true;
 	}
 
-	if (modifyTask)
+	if (updateFromQuery)
+	{
+		DeferredErrorMessage *planningError = NULL;
+		planningError = ModifyQuerySupported(query);
+
+		if (planningError != NULL)
+		{
+			multiPlan->planningError = planningError;
+			return multiPlan;
+		}
+
+		task = RouterSelectTask(originalQuery, restrictionContext, &placementList);
```

Comment:
Is it possible to send all UPDATE (and maybe DELETE) logic through this function? (which should probably be renamed)